toolkit/tools/scheduler/schedulerutils/initializegraph.go

```diff
@@ -16,6 +16,45 @@ const (
 	buildGoalNodeName = "PackagesToBuild"
 )
 
+func ReplaceRunNodesWithPrebuiltNodes(pkgGraph *pkggraph.PkgGraph, goalNode *pkggraph.PkgNode, packagesToBuild []*pkgjson.PackageVer) (err error) {
+	for _, node := range pkgGraph.AllNodes() {
+
+		//TODO: Update Build nodes to UpToDate if it is not part of packagesToBuild
```

Comment:
There is an edge case with dynamic dependencies where dangling BuildRequires might still exist. The scheduler tries to prune everything it doesn't need in `InitializeGraph()`, so if there are nodes left here which are still outstanding, and are not listed in packagesToBuild, its because we couldn't resolve all the dependencies with the static data we have (ie need to build the world and hope one of the packages gives us what we want).